test/unit/iterator/src/ConstantIterator.cpp

```diff
@@ -0,0 +1,101 @@
+/* Copyright 2021 Anton Reinhard
+ *
+ * This file is part of vikunja.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <vikunja/mem/iterator/ConstantIterator.hpp>
+
+#include <alpaka/alpaka.hpp>
+
+#include <catch2/catch.hpp>
+
+using namespace vikunja::mem::iterator;
+
+TEMPLATE_TEST_CASE("Test Constant Iterator", "", int, double, uint64_t)
```

Comment:
Please add also a test case to `integ/transform`. I think `int operator()(int i) { return 2*i;}` is a good functor for certificate the result (`sum = size * 2 * constant`).